substrate/frame/system/src/lib.rs

```diff
@@ -1091,16 +1118,17 @@ pub mod pallet {
 		type Call = Call<T>;
 		fn validate_unsigned(_source: TransactionSource, call: &Self::Call) -> TransactionValidity {
 			if let Call::apply_authorized_upgrade { ref code } = call {
-				if let Ok(hash) = Self::validate_authorized_upgrade(&code[..]) {
+				if let Ok(res) = Self::validate_code_is_authorized(&code[..]) {
```

Comment:
If a migration is ongoing, the call will fail, and here we don't test for it. But at the same time, if a migration is ongoing it is impossible to dispatch the transaction, so we are fine.

Reply:
The call is still valid. We are not testing here if the call will not error, we never do this.

Reply:
Yes, but failing code is roll-backed, so people can spam the chain with this failing code for free, it is better to avoid.
But here we are fine.
And also in any case, the provides being unique, I think it can't spam the transaction pool, only one per block would then be included.

Reply:
You should not upload invalid wasm files that would fail this check :) 

Also the code is now changed to remove the authorization if someone tried and the version check failed. So, you can only send this once.

Reply:
the check I was talking about is `T::MultiBlockMigrator::ongoing()` https://github.com/paritytech/polkadot-sdk/blob/007565a7225cba59d236ffc2fdaca2438a3665c1/substrate/frame/system/src/lib.rs#L2269-L2271

Not about the version check.

Reply:
Hmm yeah okay, good point.